compiler/resolution/lowerIterators.cpp

```diff
@@ -1290,8 +1302,16 @@ expandIteratorInline(ForLoop* forLoop) {
     // the yielded index for the iterator formal.
     expandBodyForIteratorInline(forLoop, ibody, index);
 
-    collect_asts(ibody, asts);
 
+    if (iterator->throwsError()) {
+      // For an inlined iterator, we don't want to track the error
+      // argument through the iterator class. Instead of setting
+      // a formal error argument, set an enclosing error variable.
+      collect_asts(ibody, asts);
+      replaceErrorArgumentWithEnclosingError(iterator, asts);
+    }
+
+    collect_asts(ibody, asts);
     replaceIteratorFormalsWithIteratorFields(iterator, ic, asts);
```

Comment:
First, replaceErrorArgumentWithEnclosingError and replaceIteratorFormalsWithIteratorFields need only SymExprs. So I'd like 'asts' to be a vector of SymExprs, not all BaseASTs.

Second, replaceErrorArgumentWithEnclosingError does not change the set of SymExprs. So collect_asts or collectSymExprs should be done just once.

If either of the above is not the case, please ignore.